packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/armstreamstyleserialization/models/FunctionProperties.java

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+
+package tsptest.armstreamstyleserialization.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+import java.io.IOException;
+import tsptest.armstreamstyleserialization.fluent.models.FunctionConfiguration;
+
+/**
+ * The FunctionProperties model.
+ */
+@Fluent
+public class FunctionProperties implements JsonSerializable<FunctionProperties> {
+    /*
+     * Discriminator property for FunctionProperties.
+     */
+    private String kind = "FunctionProperties";
+
+    /*
+     * The properties property.
+     */
+    private FunctionConfiguration innerProperties = new FunctionConfiguration();
+
+    /**
+     * Creates an instance of FunctionProperties class.
+     */
+    public FunctionProperties() {
+    }
+
+    /**
+     * Get the kind property: Discriminator property for FunctionProperties.
+     * 
+     * @return the kind value.
+     */
+    public String kind() {
+        return this.kind;
+    }
+
+    /**
+     * Get the innerProperties property: The properties property.
+     * 
+     * @return the innerProperties value.
+     */
+    FunctionConfiguration innerProperties() {
```

Comment:
codegen change for parent class, changing getter from private -> package private if all polymorphic models are in the same package.